### ISSUE TITLE
How to ensure fairness for multiple queues consumed by a single worker?

### ISSUE THREAD

**Issue by asma-kothari** [celery/celery#2192]
Consider 2 queues being consumed by a worker:

celery worker --app=<app> --queues=queueA,queueB

The worker is expected to guarantee fairness, that is, it should work in a round robin fashion, picking up 1 task from queueA and moving on to another to pick up 1 task from the next queue that is queueB, then again from queueA, hence continuing this regular pattern.

How can we ensure that the worker is fair with both the queues without setting CELERYD_PREFETCH_MULTIPLIER = 1?

CELERYD_PREFETCH_MULTIPLIER is set as default (that is 4) while concurrency is given a value = 8

**Comment by nezda** [comment]
http://docs.celeryproject.org/en/latest/userguide/optimizing.html#id4 says RabbitMQ (now?) delivers messages round-robin - has this changed since https://github.com/celery/celery/issues/2192#issuecomment-51799123 or are the docs wrong?

**Comment by jarshwah** [comment]
For future visitors, the order that celery consumes tasks from when setup to consume from multiple queues seems to depend on the broker **library** not the backend (rabbitmq vs redis isn't the issue).

I ran some tests and posted the results to stackoverflow: https://stackoverflow.com/a/61612762/10583

The pyamqp library pointing at rabbitmq 3.8 processed multiple queues in round-robin order, not fifo.

I haven't done any testing with redis or the older rabbitmq connector to verify other libraries behave differently.

**Comment by mrn-aglic** [comment]
It doesn't seem to be round-robin for Redis broker. 
I have three workers running with a periodic add task setup to execute each minute. 
This is what flower shows: 

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/2858231/190921860-36527eaa-c9b5-4f0b-a1b8-abd08209597f.png">


Here is the task setup:
```
@app.on_after_finalize.connect
def schedule_tasks(sender, **kwargs):
    sender.add_periodic_task(
        crontab(minute="*/1"),
        add.s(2, 3),
        name="add_numbers"
    )
```

I'm running everything in Docker, with celery 5.2.7.

**Comment by mrn-aglic** [comment]
I also tried the following: I have two apply async tasks defined while module is loading. 
I also sent a few task using flower async-apply endpoint.


Neither method seems to result in round-robin behaviour for workers.

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/2858231/190923366-a9cafec6-6e0b-4583-9fcf-0b0581418e39.png">


I also tried to define 9 tasks using apply_async in the module:

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/2858231/190923433-3a658d0a-d467-4266-b5b0-756b2ce9b5d4.png">
